### PR TITLE
Fix for RPi 3

### PR DESCRIPTION
**Pull request by AaronDewes**
Should help #64. Because I don't use Umbrel, and I don't own a RPi 3, I can't test this.
@qnjctckzz8 Can you test this on your Pi 3 when the image is build?
Also, I would need someone to test on RPi4.
I chose to test for RPi 4 because it only runs on 3 and 4, and I couldn't find a check to detect 3B and 3B+ (but I don't know much about regex).
The check was taken from raspi-config.

### CODE REVIEW

**Comment by mayankchhabra** [comment]
Wow, your super fast @AaronDewes! Concept ACK.

Ideally, we want to keep Umbrel OS as dumb and deprived of logic as possible, and have the logic sit inside the main repo - [getumbrel/umbrel](https://github.com/getumbrel/umbrel) instead. It helps with code portability, maintenance, and most importantly - allowing us to update the logic on the fly via OTA updates.

So you can instead move this logic to [`configure-box.sh`](https://github.com/getumbrel/umbrel/blob/0277601d175e1c49323b7d36e277fa93d20f192b/configure-box.sh#L89) in getumbrel/umbrel.

In fact, you can even skip checking for RPi specifically and make `dbcache` and `maxmempool` smaller in proportion to the host's RAM size.

**Comment by AaronDewes** [comment]
> In fact, you can even skip checking for RPi specifically and make dbcache and maxmempool smaller in proportion to the host's RAM size.

How small?
On RPi 4 4GB it is about 1/4 (dbcache) and 1/8 (maxmempool), for best performance in #64 both were 1/16

**Comment by mayankchhabra** [comment]
That's a good question.

cc @lukechilds

**Comment by lukechilds** [comment]
I don't think mempool is actually used at all during IBD and dbcache is allowed to overflow into any unused mempool space.

So we basically get both `dbcache` and `maxmempool` values combined to use for dbcache during IBD.

However I don't think it's particularly useful for a normal node to have `maxmempool` larger than the default 300MB. It's probably more of an inconvenience to be aware of certain mempool transactions that the majority of node's aren't since then you're more likely to be out of sync with which UTXOs are available to be spent.

So I think we should keep `maxmempool` at the default 300 and increase `dbcache` to aim for 50% RAM utilisation for both combined.

So basically:

```
maxmempool = 300
dbcache = (total_ram / 2) - 300
```

So we'd get:

1GB
`maxmempool` = 300
`dbcache` = 200

2GB
`maxmempool` = 300
`dbcache` = 700

4GB
`maxmempool` = 300
`dbcache` = 1700

8GB
`maxmempool` = 300
`dbcache` = 3700

In the future I think we can set `dbcache` even higher for IBD (maybe 75%-80% total RAM) and then reduce to a lower value once IBD is complete (https://github.com/getumbrel/umbrel/issues/39) but the above values seem like a sensible default for now.

**Comment by nolim1t** [comment]
can confirm mempool isn't used while it is syncing

**Comment by AaronDewes** [comment]
This can be optimized, but I don't know much about sed.

1/2 of Total RAM:
```bash
expr $(grep MemAvailable /proc/meminfo | sed 's/MemAvailable://g' | sed -e 's/^[ \t]*//' | sed 's/kB//g') / 2048
``` 
Alternatively, 2/3 of available RAM would be possible.
```bash
expr $(grep MemAvailable /proc/meminfo | sed 's/MemAvailable://g' | sed -e 's/^[ \t]*//' | sed 's/kB//g') / 1536
```

**Comment by meeDamian** [comment]
@lukechilds LGTM

@AaronDewes Two above can also be written as:

```sh
awk '/MemTotal/{printf "%d\n", $2/2^10 * 0.5}'  /proc/meminfo
awk '/MemTotal/{printf "%d\n", $2/2^10 * 0.75}' /proc/meminfo
```

And if decimal point values are alright:

```
awk '/MemTotal/{print $2/2^10 * 0.5}'  /proc/meminfo
awk '/MemTotal/{print $2/2^10 * 0.75}' /proc/meminfo
```

**Comment by AaronDewes** [comment]
Thank you. However, I used 1/2 of the total RAM and 2/3 of the available RAM. (Available documentation: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773).

@lukechilds What should I use?
```Bash
awk '/MemTotal/{printf "%d\n", $2/2^10 * 0.5}'  /proc/meminfo  # 50% of total mem (in MB)
awk '/MemAvailable/{printf "%d\n", $2/2^10 * 0.75}' /proc/meminfo  # 75% of avail mem (in MB)
```

**Comment by lukechilds** [comment]
>Thank you. However, I used 1/2 of the total RAM and 2/3 of the available RAM. (Available documentation: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773).
>
>@lukechilds What should I use?

Sorry, should've been more clear. By available RAM I meant total amount on the system, not just the amount that's free. So we treat a 1GB Pi as having 1GB available RAM (so maxmempool = 300/dbcache = 200) even if the system is already taking up a few hundred MB.

>>And if decimal point values are alright
>
>Does bitcoin.conf support them?

I don't think so, lets stick to integers, it doesn't need to be super accurate.
